packages/video_player/video_player_platform_interface/lib/video_player_platform_interface.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart' show visibleForTesting;
+
+import 'method_channel_video_player.dart';
+
+/// The interface that implementations of video_player must implement.
+///
+/// Platform implementations should extend this class rather than implement it as `video_player`
+/// does not consider newly added methods to be breaking changes. Extending this class
+/// (using `extends`) ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by newly added
+/// [VideoPlayerPlatform] methods.
+abstract class VideoPlayerPlatform {
+  /// Only mock implementations should set this to true.
+  ///
+  /// Mockito mocks are implementing this class with `implements` which is forbidden for anything
+  /// other than mocks (see class docs). This property provides a backdoor for mockito mocks to
+  /// skip the verification that the class isn't implemented with `implements`.
+  @visibleForTesting
+  bool get isMock => false;
+
+  /// The default instance of [VideoPlayerPlatform] to use.
+  ///
+  /// Platform-specific plugins should override this with their own
+  /// platform-specific class that extends [VideoPlayerPlatform] when they
+  /// register themselves.
+  ///
+  /// Defaults to [MethodChannelVideoPlayer].
+  static VideoPlayerPlatform _instance = MethodChannelVideoPlayer();
+
+  static VideoPlayerPlatform get instance => _instance;
+
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(VideoPlayerPlatform instance) {
+    if (!instance.isMock) {
+      try {
+        instance._verifyProvidesDefaultImplementations();
+      } on NoSuchMethodError catch (_) {
+        throw AssertionError(
+            'Platform interfaces must not be implemented with `implements`');
+      }
+    }
+    _instance = instance;
+  }
+
+  /// Initializes the platform interface and disposes all existing players.
+  ///
+  /// This method is called when the plugin is first initialized
+  /// and on every full restart.
+  Future<void> init() {
+    throw UnimplementedError('init() has not been implemented.');
+  }
+
+  /// Clears one video.
+  Future<void> dispose(int textureId) {
+    throw UnimplementedError('dispose() has not been implemented.');
+  }
+
+  /// Creates an instance of a video player.
+  ///
+  /// For network and file sources [dataSourceDescription] needs to have a
+  /// [uri] key. Optionally you can set a [formatHint].
+  ///
+  /// For assets the [asset] key is mandatory. Optionally you can specify
+  /// the [package].
+  /// TODO (cbenhagen): create a [DataSource] class
```

Comment:
You'll want to create the `DataSource` class in this PR, because you will have to update the platform interface package and publish it to pub before the `video_player` package can use it.

Reply:
Ok. Will do that tomorrow. Super late here in Switzerland already.

Reply:
Done